<center><img src="logo_ilum_2023.png" style="width:1000px;height:200px"/></center>

# 🥼 Caderno de experimentos de Termodinâmica Avançada 

## 🧪 Desafio 1

**Objetivo:** A definição de uma mistura com potencial para fornecimento de energia ao ciclo Rankine (responsável pela geração de energia elétrica).

### Escolha da mistura

<p style='text-align: justify'>
    Para início da justificativa de escolha de uma mistura eficiente para o desenvolvimento do trabalho, é necessário o entendimento do ciclo de Rankine. Esse ciclo termodinâmico apresenta uma eficiente capacidade de extrair a energia térmica de fluídos e, posteriormente, obter a partir deles energia mecânica em grande quantidade. O funcionamento do ciclo consiste na conversão do estado líquido para de vapor o que movimenta uma turbina, ou seja, gerando energia mecânica, e depois ele retorna ao seu estado líquido com pouca energia sendo removida pelos condensadores no mecanismo. Então, tendo em vista este ciclo, a escolha da mistura deveria considerar uma mistura com grande quantidade de energia, além disso, outro fator importante para essa consideração seria o seu impacto para o meio ambiente. Sendo assim, decidimos utilizar o biocombustível de alga devido ao fato de seu óleo ser rico em energia, ao seu fator de alta proliferação em curto período de tempo e também pela sua capacidade de realizar fotossíntese e contribuir para o meio ambiente sem poluí-lo.</p>
    
<p style='text-align: justify'>
    Ao iniciar a pesquisa sobre o nosso líquido escolhido, no entanto, tivemos dificuldade em encontrar as informações necessárias sobre ele, sua entalpia, diagrama de fases e algum outro líquido que com ele poderia compor uma boa e realística mistura. Dessa forma, passamos a buscar outras substâncias que seria bom para o nosso estudo durante o curso, considerando por hora apenas a disponibilidade de dados sobre essa possível mistura além de suas propriedades termodinâmicas. Com isso, encontramos a solução de p-Xileno com 1-Hexeno e o escolhemos devido ao fato do p-xileno ser altamente inflamável e muito utilizado na indústria. Ademais, o ponto de ebulição dos dois também foram considerados e sendo eles de 138°C e 63,4°C ideias para o uso no ciclo de Rankine.</p>

### Entalpia

<p style='text-align: justify'>
    Pensando no objetivo de escolher uma mistura com potencial para o fornecimento de energia ao ciclo de Rankine, que é responsável pela geração de energia elétrica, uma propriedade muito importante que consideramos foi a <b>entalpia</b>.</p>

<p style='text-align: justify'>
    Ela pode ser compreendida como a quantidade de calor ou energia que é necessária colocar para criar o sistema em questão a partir do nada e inseri-lo no ambiente ou, ainda, a energia que é preciso extrair para aniquilar o sistema. Assim, ela é dada como uma quantidade total de energia requerida e fornecida, adicionando PV na energia U:</p>

<p style='text-align: center'><b>H = U + PV</b> ,</p>

<p style='text-align: justify'> sendo PV = W, trabalho, que é considerado visto a necessidade de energia para tirar o ar atmosférico do lugar para o sistema passar ocupá-lo ou a que é extraída, para além de U, quando ele colapsar.</p>

<p style='text-align: justify'>
    Supondo que algumas mudanças, como adição de calor, fossem feitas à pressão constante, seria possível observar as seguintes variações: ΔV, ΔU e ΔH. Com isso, </p>
    
<p style='text-align: center'>H + ΔH = (U + ΔU) + P(V + ΔV)</p>
<p style='text-align: center'>H + ΔH = (U + PV) + (ΔU + PΔV)</p>
<p style='text-align: center'>H + ΔH = H + (U + PV) + (ΔU + PΔV)</p>
<p style='text-align: center'><b>ΔH = ΔU + PΔV</b></p>

<p style='text-align: justify'>
    Dessa definição, podemos concluir que a entalpia vai aumentar quando a energia (U) aumentar ou quando o sistema expandir, realizando trabalho (PV) na atmosfera.</p>
    
<p style='text-align: justify'>
    Considerando a primeira lei da termodinâmica, </p>

<p style='text-align: center'>ΔU = Q - PΔV ,</p>

<p style='text-align: justify'> sendo a pressão constante, ΔU a variação de energia feita, Q a quantidade de calor adicionado no sistema e PΔV o trabalho feito de compressão-expansão, além de considerar que não há outro tipo de trabalho sendo feito, realocando essa definição na fórmula de variação de entalpia encontrada, temos:</p>

<p style='text-align: center'>ΔH = (Q - PΔV) - PΔV</p>
<p style='text-align: center'><b>ΔH = Q.</b></p>

<p style='text-align: justify'>
    O que nos leva a definição operacional de entalpia em que, à pressão constante e sem outros tipos de trabalho, a variação na entalpia é equivalente, indicando diretamente, o quanto de calor foi adicionado ao sistema, em outras palavras, o quanto de calor se tem no sistema que está imerso nessa vizinhança de pressão constante.</p>
    
<p style='text-align: justify'>
    Sobre a entalpia, ainda há dois conceitos que usamos para a determinação da nossa mistura. O primeiro é <b>entropia de formação</b>, Δ<sub>f </sub>H<sup>o</sup> , que diz respeito a entalpia necessária para a formação de um mol um composto a partir de seus elementos fundamentais em seus estados estáveis. O segundo é a propriedade de <b>aditividade</b>, conhecida por <b>Lei de Hess</b>, a qual estabelece que, se somarmos duas equações químicas, a fim de se obter uma terceira equação química, o valor de Δ<sub>r </sub>H (entalpia da reação) para a equação resultante é igual a soma de Δ<sub>r </sub>H das duas equações sendo somadas. Com isso,<p>
        
<p style='text-align: center'>Δ<sub>r </sub>H(3) = <sub>r </sub>H(1) + <sub>r </sub>H(2)</p>
<p style='text-align: center'>Δ<sub>r </sub>H = Δ<sub>r </sub>H (inversa).</p>

<p style='text-align: justify'>
    Dessa forma, encontramos que </p>
    
<p style='text-align: center'>Xylene: Δ<sub>f </sub>H<sup>o</sup>(g) = 17.9 kJ/mol , com imprecisão de 1</p>
<p style='text-align: center'>1-Hexene: Δ<sub>f </sub>H<sup>o</sup>(g) = -42 kJ/mol , com imprecisão de 2.</p>
    
<p style='text-align: justify'>
    Assim, </p>
    
<p style='text-align: center'>Δ<sub>r </sub>H = -24.1 kJ/mol, com erro de 3.</p>
    
<p style='text-align: justify'>
    Logo, por se caracterizar como uma reação exotérmica, em que Δ<sub>r </sub>H é menor que 0, ou seja, libera calor ao sistema, é uma boa escolha de mistura para nosso objetivo</p>

### Diagrama de fases

<p style='text-align: justify'>
    O diagrama de fases da mistura de p-xileno e 1-hexeno foi encontrado no banco de dados Dortmund. Nele está representado no eixo das ordenadas a temperatura em Kelvin e no eixo das abscissas a porcentagem de 1-hexeno na solução.</p>

<center><img src="diagrama_de_fases.png" style="width:600px;height:600px"/></center>

----------

## ⚗ Desafio 2

**Objetivo:** Separação dos líquidos na mistura e a evaporação do combustível.

-----
_Aspectos importantes que devem ser considerados:_

>O volume final da mistura calculado a partir dos volumes parciais molares.

>O comportamento da mistura de líquido (ideal ou não ideal)

>As características do diagrama de fases da mistura líquida.

>A pressão de vapor dos compostos
</p>

----------

<p style='text-align: justify'> Primeiramente, é necessário calcular o volume total da mistura para se entender como que o sistema irá se comportar nas próximas etapas. Para realizar o cálculo, haja visto que o volume é um parâmetro extensivo,partimos da equação do volume total por volumes parciais molares:</p>

<p style='text-align: center'>
V = V<sub>p1</sub>.n<sub>1</sub> + V<sub>p2</sub>.n<sub>2</sub> .
</p>

<p style='text-align: justify'> Dividindo pelo número total de mols:</p>

<p style='text-align: center'>
V/(n<sub>1</sub> + n<sub>2</sub>) = V<sub>p1</sub>.n<sub>1</sub>/(n<sub>1</sub> + n<sub>2</sub>) + V<sub>p2</sub>.n<sub>2</sub>/(n<sub>1</sub> + n<sub>2</sub>)
</p>
<p style='text-align: center'>
V<sub>(por mol)</sub> = V<sub>p1</sub>.x<sub>1</sub> + V<sub>p2</sub>.x<sub>2</sub> ,
</p>
<p style='text-align: center'>
em que x1 e x2 são a fração molar do composto 1 e a do 2, respectivamente.
</p>

<p style='text-align: justify'> Agora, utilizando a relação da densidade, temos que:</p>

<p style='text-align: center'>
 -> Sabendo x<sub>1</sub> e tendo em mente que x<sub>1</sub> + x<sub>2</sub> = 1 (complementares): 
</p>
<p style='text-align: center'>
ρ<sub>mistura</sub>(x<sub>1</sub>) = MW<sub>1</sub> / V<sub>p1</sub>(x<sub>1</sub>)
</p>
<p style='text-align: center'>
ρ<sub>mistura</sub>(x<sub>1</sub>) = MW<sub>2</sub> / V<sub>p2</sub>(x<sub>1</sub>)
</p>
<p style='text-align: center'>
 -> Substituindo na equação de volume: 
</p>
<p style='text-align: center'>
V<sub>(por mol)</sub> = (V<sub>p1</sub>.x<sub>1</sub>.MW<sub>1</sub>)/ρ<sub>mistura</sub>(x<sub>1</sub>) + (V<sub>p2</sub>.x<sub>2</sub>MW<sub>2</sub>)/ρ<sub>mistura</sub>(x<sub>1</sub>)
</p>
<p style='text-align: center'>
 -> Reescrevendo, lembrando da relação de complementariedade de x<sub>1</sub> e x<sub>2</sub>: 
</p>
<p style='text-align: center'>
V<sub>M</sub><sub>(por mol)</sub> = (V<sub>p1</sub>.x<sub>1</sub>.MW<sub>1</sub> + V<sub>p2</sub>.(1 - x<sub>1</sub>)MW<sub>2</sub>)/ρ<sub>mistura</sub>(x<sub>1</sub>) ,
</p>
<p style='text-align: center'> sendo MW<sub>1</sub> e MW<sub>2</sub> os pesos moleculares (molar weights) de cada composto, respectivamente.</p>

<p style='text-align: justify'> O que nos permite calcular, se tivermos valores diferentes para cada densidade da mistura, medidas experimentalmente para cada fração molar, o volume por mol da mistura.</p>

----

In [8]:
#Importando bibliotecas necessárias para o desafio

import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

_Definindo valores:_

<p style='text-align: justify'> Massas em gramas por mol:</p>

In [9]:
mmol_p_xylene = 106.16  # massa molar [g/mol]: p-xileno
mmol_1_hexene = 84.16   # massa molar [g/mol]: 1-hexeno

<p style='text-align: justify'> Densidade da mistura a depender da fração (a fração molar considerada é a do hexano):</p>

In [10]:
ρ_50 = 0.76  # densidade [g/cm3] para 0.5 do 1-hexano para 1 mol da mistura
ρ_63 = 0.74  # densidade [g/cm3] para 0.63 do 1-hexano para 1 mol da mistura
ρ_73 = 0.72  # densidade [g/cm3] para 0.73 do 1-hexano para 1 mol da mistura
ρ_84 = 0.70  # densidade [g/cm3] para 0.84 do 1-hexano para 1 mol da mistura
ρ_93 = 0.68  # densidade [g/cm3] para 0.93 do 1-hexano para 1 mol da mistura

-----

Para o nosso problema temos como definição que a massa final da mistura deve ser de 1 tonelada, sendo assim, algumas conversões são necessárias para o cálculo do volume final, as quais estão descritas a baixo

In [13]:
massa_total = 1e6 #g

<p style='text-align: justify'> Função para o cálculo do volume por mol da mistura:</p>

In [11]:
def volume_por_mol(ρ,X1,MW1,MW2):
    
    
    Volume_pmol = ((X1*MW1)+((1-X1)*MW2))/ρ
    
    return Volume_pmol 

In [12]:
print(volume_por_mol(ρ_63,0.63,mmol_1_hexene,mmol_p_xylene))

124.72972972972973


<p style='text-align: justify'> Portanto, <b>V<sub>mistura<sub>por mol</sub></sub> = 124.72972972972973 g/mol</b>.</p>

------------

<p style='text-align: justify'> Visando separar os líquidos da mistura, passamo-na por um processo de destilação fracionada simples, condensando e recondensando o vapor obtido diversas vezes, em prol de se obter um componente puro. 
    
Nesse contexto, podemos fazer a seguinte análise do diagrama de fases:
    
- Determinamos que vamos iniciar com uma fração da mistura de 0.63530 mol de 1-Hexene líquido (na temperatura de 349.98°); </p>

<center><img src="diagrama_de_fases_concentracao.png" style="width:600px;height:600px"/></center>

<p style='text-align: justify'> 
    
- Com isso, de acordo com o gráfico, temos que a concentração no vapor que vamos obter é de 0.94340 de 1-Hexene na sua forma de vapor;
    
- Com ele, condensando-o e evaporando novamente, vamos ter um vapor com a concentração de, aproximadamente 0.0.99530 de 1-Hexene;
    
- Passando a uma temperatura menor que a anterior (336.64 K), é possível obtermos, enfim, a concentração de 1 mol/mol de 1-Hexene. Ou seja, conseguimos separar a mistura, obtendo o composto 1-Hexene puro.</p>

-----------

<p style='text-align: justify'> Tendo analisado o volume, nos cabe analisar como se dá o comportamento desta mistura, verificando se é ideal ou não e analisando-o. A priori, que tal começarmos definindo o que seria cada um desses tipos de mistura?</p>

<p style='text-align: justify'> Uma <b>mistura ideal</b> se caracteriza desta maneira quando os compostos envolvidos não possuem preferências de vizinho, possuem tamanho de suas partículas praticamente iguais, compondo uma mistura bem homogênea.

Enquanto em uma <b>mistura não-ideal</b>, as moléculas têm preferência para ter como vizinhas as moléculas iguais.

Pensando nisso, quando ela é ideal, já que existe um equilíbrio natural, é possível tirar várias informações a partir de modelagens e análises em relação ao vapor, que permitem inferir sobre propriedades das substâncias, da mistura líquida.

Assim, quando ela é ideal, obedece à lei de Rault para todas as concentrações dos compostos, enquanto para não ideal, é necessário analisar seu gráfico por partes, sendo a mais próxima do idela com a lei de Raoult, outra com de Henry e o meio com modelagem.</p>

<p style='text-align: center'> <b>Lei de Raoult</b>: P<sub>1</sub> = x<sub>1</sub> . P<sub>1</sub><sob>*</sob>; P<sub>2</sub> = x<sub>2</sub> . P<sub>2</sub><sob>*</sob>; P = P<sub>1</sub> + P<sub>2</sub></p>
    
<p style='text-align: center'> <b>Lei de Henry</b>: P<sub>1</sub> = k<sub>H, 1</sub> . x<sub>1</sub>; P<sub>2</sub> = k<sub>H, 2</sub> . x<sub>2</sub>.

------------

<p style='text-align: justify'> Então, olhando para gráficos de pressão de vapor, podemos definir, por meio de alguns exemplos, como ele seria caso estivéssemos olhando para um de uma mistura ideal e também para outro não-ideal.</p>

<p style='text-align: justify'> Para a nossa mistura (1-hexene + p-xylene), não encontramos gráfico de pressão de vapor pela fração dos compostos. Logo, vamos considerá-la ideal a fim de realizar os cálculos do problema. Sendo assim, a mistura, agora considerada ideal, respeita a lei de Raoult em todas as suas concentrações.

Lembrete - tópicos a falar

- colocar o gráfico de pressao de vapor

- indicar pq é não-ideal

- indicar qual parte analisa com qual lei

- indicar gráfico de miscibilidade

- indicar como ocorre a transformação líquido-vapor, vapor-líquido de acordo com o diagrama de fases

- calcular valores da pressão de vapor
 
gráfico na temp de ebulição; conseguir pressão na temperatura e variair x = gráfico
